platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -33,70 +33,77 @@
 public abstract class AbstractLongList<C> implements LongList {
 
     public static final String MAX_CHUNKS_EXCEEDED_MSG = "The maximum number of memory chunks should not exceed %s. "
-            + "Either increase numLongsPerChunk or decrease maxLongs";
+            + "Either increase longsPerChunk or decrease capacity";
+    public static final String CHUNK_SIZE_ZERO_OR_NEGATIVE_MSG = "Cannot store %d per chunk (min is 1)";
     public static final String CHUNK_SIZE_EXCEEDED_MSG = "Cannot store %d per chunk (max is %d)";
     public static final String INVALID_RANGE_MSG = "Invalid range %d - %d";
     public static final String MAX_VALID_INDEX_LIMIT = "Max valid index %d must be less than max capacity %d";
 
-    /** A suitable default for the maximum number of longs that may be stored (32GB of longs). */
-    protected static final long DEFAULT_MAX_LONGS_TO_STORE = 4_000_000_000L;
-    /** The maximum number of longs to store per chunk. */
-    protected static final int MAX_NUM_LONGS_PER_CHUNK = toIntExact(16_000L * (MEBIBYTES_TO_BYTES / Long.BYTES));
-    /** A suitable default for the number of longs to store per chunk. */
-    protected static final int DEFAULT_NUM_LONGS_PER_CHUNK = toIntExact(8L * (MEBIBYTES_TO_BYTES / Long.BYTES));
-
-    /** A suitable default for the reserved buffer length that the list should have before minimal
-     * index in the list
+    /**
+     * The maximum number of longs to store per chunk. The value is to make sure a single chunk is
+     * less than 2Gb - there is no specific requirement about it, just looks reasonable.
      */
-    public static final int DEFAULT_RESERVED_BUFFER_LENGTH = toIntExact(2L * MEBIBYTES_TO_BYTES / Long.BYTES);
+    protected static final int MAX_NUM_LONGS_PER_CHUNK = toIntExact(2_000L * (MEBIBYTES_TO_BYTES / Long.BYTES));
 
-    /** Maximum number of chunks allowed.*/
-    public static final int MAX_NUM_CHUNKS = 2 << 14;
+    /**
+     * The maximum number of chunks.
+     */
+    public static final int MAX_NUM_CHUNKS = 2 << 20;
```

Comment:
This limit is artificial. I suggest removing it. Heap consumption from the array of chunks is negligible compared to chunks themselves.

Reply:
I left it as is. This is not about memory consumption, this is just a sanity check that we don't have too many chunks. May be removed in the future, if needed